src/db.rs

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) 2022 IBM
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::policy;
+use crate::request;
+
+use anyhow::*;
+use mysql::{OptsBuilder, Pool, PooledConn, TxOpts};
+use uuid::Uuid;
+use std::env;
+use log::*;
+use std::result::Result::Ok;
+
+use mysql::prelude::*;
+
+#[derive(Debug, Clone)]
+pub struct Connection {
+    pub policy: u32,
+    pub fw_api_major: u32,
+    pub fw_api_minor: u32,
+    pub fw_build_id: u32,
+    pub launch_description: String,
+    pub fw_digest: String,
+}
+
+impl Default for Connection {
+    fn default() -> Connection {
+        Connection {
+            policy: 0x0,
+            fw_api_major: 0,
+            fw_api_minor: 0,
+            fw_build_id: 14,
+            launch_description: "test".to_string(),
+            fw_digest: "placeholder".to_string(),
+        }
+    }
+}
+
+
+pub fn get_dbconn()-> mysql::Result<PooledConn> {
+
+    let host_name = env::var("KBS_DB_HOST").expect("KBS_DB_HOST not set");
+    let user_name = env::var("KBS_DB_USER").expect("KBS_DB_USER not set.");
+    let db_pw = env::var("KBS_DB_PW").expect("KBS_DB_PW not set.");
+    let data_base = env::var("KBS_DB").expect("KBS_DB not set");
+
+    let opts = OptsBuilder::new()
+        .ip_or_hostname(Some(host_name))
+        .user(Some(user_name))
+        .pass(Some(db_pw))
+        .db_name(Some(data_base));
+
+    let dbpool = Pool::new(opts).unwrap();
+    let dbconn = dbpool.get_conn().unwrap();
+
+    Ok(dbconn)
+}
+
+
+pub fn insert_connection(connection: Connection) -> Result<Uuid> {
+
+    let mut dbconn = get_dbconn().unwrap();
+
+    let nwuuid = Uuid::new_v4();
+    let uuidstr = nwuuid.to_hyphenated().to_string();
+
+    let mqstr = "INSERT INTO conn_bundle (id, policy, fw_api_major, fw_api_minor, 
+                 fw_build_id, launch_description, fw_digest,create_date) 
+                 VALUES(?, ?, ?, ?, ?, ?, ?,NOW())".to_string();
+
+    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+
+    trnsx.exec_drop(
+        mqstr,
+        (
+            uuidstr,
+            connection.policy,
+            connection.fw_api_major,
+            connection.fw_api_minor,
+            connection.fw_build_id,
+            connection.launch_description,
+            connection.fw_digest,
+        ),
+    )?;
+    trnsx.commit()?;
+    Ok(nwuuid)
+}
+
+pub fn delete_connection(uuid: Uuid) -> Result<Uuid> {
+
+    let mut dbconn = get_dbconn().unwrap();
+
+    let mqstr ="DELETE from conn_bundle WHERE id = ?";
```

Comment:
This connection table is more for future stuff and will get rehashed a bit. Currently we store the connection bundles in a hash table (because you can't serialize the SEV Session struct). We also probably won't be using this delete function aside from testing. Instead, as you suggest, it will be up to the guest owner to occasionally prune old entries